src/app/(proper_react)/(onboarding)/user/welcome/EnterInfo.tsx

```diff
@@ -4,22 +4,322 @@
+    >
+      <p>{l10n.getString("onboarding-enter-details-comfirm-dialog-text")}</p>
+      <div className={viewStyles.dialogContents}>
+        <ul className={enterInfoStyles.infoList}>
+          {userDetailsData.map(({ key, label, displayValue }) => (
+            <li key={key} className={enterInfoStyles.infoItem}>
+              {label}: <strong>{displayValue}</strong>
+            </li>
+          ))}
+        </ul>
+      </div>
+      <div className={viewStyles.stepButtonWrapper}>
+        <Button
+          type="secondary"
+          onClick={() => confirmDialogState.close()}
+          className={enterInfoStyles.startButton}
+        >
+          {l10n.getString(
+            "onboarding-enter-details-comfirm-dialog-button-edit"
+          )}
+        </Button>
+        <Button
+          type="primary"
+          onClick={() => props.onDataSaved()}
+          autoFocus={true}
+          className={enterInfoStyles.startButton}
+        >
+          {l10n.getString(
+            "onboarding-enter-details-comfirm-dialog-button-confirm"
+          )}
+        </Button>
+      </div>
+    </Dialog>
+  );
 
   return (
-    <div className={styles.wrapper}>
-      Enter your info here. After it&apos;s sent to the backend, we&apos;ll call{" "}
-      <code>props.onDataSaved</code>.
+    <div className={viewStyles.stepContent}>
+      <h1>{l10n.getString("onboarding-enter-details-title")}</h1>
+      <p>
+        {l10n.getString("onboarding-enter-details-text")}
+        <button
+          {...explainerDialogTrigger.triggerProps}
+          onClick={() => explainerDialogState.open()}
+          className={viewStyles.explainerTrigger}
+        >
+          {l10n.getString("onboarding-enter-details-why-button-label")}
+        </button>
+      </p>
+      <div className={enterInfoStyles.inputContainer}>
+        {userDetailsData.map(
+          ({
+            key,
+            errorMessage,
+            label,
+            onChange,
+            placeholder,
+            type,
+            value,
+          }) => (
+            <InputField
+              key={key}
+              errorMessage={errorMessage}
+              label={label}
+              isRequired={true}
+              onChange={onChange}
+              placeholder={placeholder}
+              type={type}
+              validationState={
+                !getIsValidInfo(value) && invalidInputs.includes(key)
+                  ? "invalid"
+                  : "valid"
+              }
+              value={value}
+            />
+          )
+        )}
+      </div>
+      <div className={viewStyles.stepButtonWrapper}>
+        <Button
+          type="secondary"
+          onClick={() => props.onGoBack()}
+          className={enterInfoStyles.startButton}
+        >
+          {l10n.getString("onboarding-steps-enter-info-back")}
+        </Button>
+        <Button
+          {...confirmDialogTrigger.triggerProps}
+          type="primary"
+          onClick={validateEnteredInfo}
```

Comment:
Could you wrap the input fields and this button in a `<form>` and set this as the `onSubmit` handler instead (and have it call `event.preventDefault()`)? Then people can submit the form by pressing Enter while having one of the input fields focused.

(If there are any other buttons inside the `<form>`, you'll have to set those to be `type="button"` to prevent them from submitting the form.)

Reply:
Done in [1b294a9](https://github.com/mozilla/blurts-server/pull/3208/commits/1b294a9bbe3570e375e9259339b2b1121b036433).